guides/02_building-interfaces/05_four-kinds-of-interfaces.md

```diff
@@ -0,0 +1,47 @@
+# The 4 Kinds of Gradio Interfaces
+
+So far, we've always assumed that in order to build an Gradio demo, you need both inputs and outputs. But this isn't always the case for machine learning demos: for example, *unconditional image generation models* don't take any input but produce an image as the output.
+
+It turns out that the `gradio.Interface` class can actually handle 4 different kinds of demos:
+
+1. **Standard demos**: which have both separate inputs and outputs (e.g. an image classifier or speech-to-text model)
+2. **Output-only demos**: which don't take any input but produce on output (e.g. an unconditional image generation model)
+3. **Input-only demos**: which don't produce any output but do take in some sort of input (e.g. a demo that saves images that you upload to a persistent external database)
+4. **Unified demos**: which have both input and output components, but the input and output components *are the same*. This means that the output produced overrides the input (e.g. a text autocomplete model)
+
+Depending on the kind of demo, the user interface (UI) looks slightly different:
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-guides/interfaces4.png)
+
+
+Let's see to build each kind of demo using the `Interface` class, along with examples:
+
+
+## Standard demos
+
+To create a demo that has both the input and the output components, you simply need to set the values of the `inputs` and `outputs` parameter in `Interface()`. Here's an example demo of a mock image classification model:
```

Comment:
Below demo is an image-to-image demo, so not really image classification at all, would just call it something else.

Reply:
Good catch